students/Sally_Shen/lesson4/fileExercise.py

```diff
@@ -0,0 +1,99 @@
+import os
+
+'''
+Paths and File Processing
+'''
+def listFiles():
+    for line in os.listdir():
+        print(os.path.abspath(line))
+
+# listFiles()
+
+
+def CopyFile(src, dest):
+    with open(src, "rb") as source:
+        with open(dest, "wb") as destination:
+            while True:
+                data = source.read(1)
+                if data:
+                    destination.write(data)
+                else:
+                    break
+
+
+#CopyFile("from.pdf", "to.pdf")
+#CopyFile("from.txt", "to.txt")
+
+
+'''
+File reading and parsing
+'''
+def GenLanguages():
+    ExistingLanguages = {
+        "python", "java", "perl", "fortran",
+        "shell", "matlab", "bash", "c#",
+        "javascript", "typescript", "c++", "c",
+        "erlang", "ansible", "powershell", "r",
+        "visualbasic", "php", "mysql", "rex",
+        "db", "vb", "lisp", "sql",
+        "someNewLanguage"}
+    LanguagesFromStudents = set()
+    with open("students.txt", "r") as f:
+        for line in f:
+            #print("line:" + line)
+            values = line.strip().split(":")
+            if not values[1]:
+                # no potential language list
+                continue
+            else:
+                potentialLanguages = values[1]
+                for potentialLanguage in potentialLanguages.split(","):
+                    potentialLanguage = potentialLanguage.strip()
+                    #print("potentialLanguage:" + potentialLanguage)
+                    if potentialLanguage in ExistingLanguages:
```

Comment:
The way to tell if the string is potential language is that the languages are all in lowercase while the nicknames are uppercase. So what you want to do is validate if potentialLanguage is capitalized (check if the first letter is upper case). 

Otherwise,  if we have a language not specified above, it will skip it.